gossipd/gossip_store.c

```diff
@@ -0,0 +1,48 @@
+#include "gossip_store.h"
+
+#include <ccan/endian/endian.h>
+#include <ccan/read_write_all/read_write_all.h>
+#include <common/status.h>
+#include <fcntl.h>
+
+#define GOSSIP_STORE_FILENAME "gossip_store"
+
+struct gossip_store {
+	int read_fd, write_fd;
+};
+
+struct gossip_store *gossip_store_new(const tal_t *ctx)
+{
+	struct gossip_store *gs = tal(ctx, struct gossip_store);
+	gs->write_fd = open(GOSSIP_STORE_FILENAME, O_RDWR|O_APPEND|O_CREAT, 0600);
+	gs->read_fd = open(GOSSIP_STORE_FILENAME, O_RDONLY);
+	return gs;
+}
+
+void gossip_store_append(struct gossip_store *gs, const u8 *msg)
+{
+	u16 msglen = tal_len(msg);
+	beint16_t belen = CPU_TO_BE16(msglen);
```

Comment:
Use `cpu_to_be16()` instead of` CPU_TO_BE16` which is designed for constants and only exists so that we can get compile-time-constants if needed.